lib/Backends/OpenCL/CMakeLists.txt

```diff
@@ -47,11 +47,16 @@ add_library(OpenCLDeviceManager
 target_link_libraries(OpenCLDeviceManager
                       PRIVATE
                         Backend
-                        Backends
+                        OpenCLBackend
```

Comment:
@rdzhabarov Do we need the authors of backends and device managers take care of something or keep an invariant when they use these new factories? E.g. is there a rule like "Do not explicitly depend on Backends, but depend on the specific backend or specific factory?" If there are such rules now, we should state it somewhere.

Reply:
Updated doc.